shop/src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java

```diff
@@ -0,0 +1,62 @@
+package ro.msg.learning.shop.controller;
+
+import lombok.NonNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.exception.ResourceNotFoundException;
+import ro.msg.learning.shop.mapper.ProductCategoryMapper;
+import ro.msg.learning.shop.service.ProductCategoryService;
+
+import java.util.List;
+import java.util.UUID;
+
+@Validated
```

Comment:
@Validated is used in order to make certain validations active in an mvc scenario.
@NonNull is a validation, but is not one which requires @Validated, therefore the @Validated is pointelss. 
Please remove and read the @Validated documentation. Actually, read any documentation before applying annotations which you might have seen in other codes (u know what i'm talking about 👀 ). The purpose of inspiring oneself from other projects is that you first investigate what you want to copy and only after the investigation you decide if you copy it or not

Reply:
Removed